packages/manager/modules/pci/src/components/project/gateways/gateways-model-selector/controller.js

```diff
@@ -1,16 +1,34 @@
 export default class PciProjectGatewayModelSelectorCtrl {
   /* @ngInject */
-  constructor(PciProjectGatewayModelSelectorService) {
+  constructor(coreConfig, PciProjectGatewayModelSelectorService) {
+    this.coreConfig = coreConfig;
     this.PciProjectGatewayModelSelectorService = PciProjectGatewayModelSelectorService;
   }
 
   $onInit() {
     this.loading = true;
+    this.currency = null;
+    this.gateway = {};
+    this.yearlyGatewayPrice = 12;
+    this.user = this.coreConfig.getUser();
     this.PciProjectGatewayModelSelectorService.getGatewayCatalog(
       this.ovhSubsidiary,
     )
       .then((data) => {
-        this.gatewayCatalog = data;
+        this.gatewayCatalog = [
+          ...new Set(data.map((gatewayInfo) => gatewayInfo.product)),
+        ].map((product) => {
+          return {
+            product,
+            gatewayInfo: data
+              .filter((gateway) => {
+                return gateway.product === product;
+              })
+              .map((gateway) => {
+                return gateway;
+              }),
```

Comment:
**nitpick(non-blocking):** If nothing is modified. What is the point of using `map`?